spec/models/post_spec.rb

```diff
@@ -1,43 +1,43 @@
-#    user = User.create(name: 'John', posts_counter: 0)
-#    post = Post.new(title: 'Test Post', comments_counter: 0, likes_counter: 0, author: user)
-#    expect(post).to be_valid
-#  end
+RSpec.describe Post, type: :model do
+  it 'is valid with a title, comments_counter, and likes_counter greater than or equal to zero' do
+    user = User.create(name: 'John', posts_counter: 0)
+    post = Post.new(title: 'Test Post', comments_counter: 0, likes_counter: 0, author: user)
+    expect(post).to be_valid
+  end
 
-#  it 'is invalid without a title' do
-#    post = Post.new(comments_counter: 0, likes_counter: 0)
-#    expect(post).to_not be_valid
-#  end
+  it 'is invalid without a title' do
+    post = Post.new(comments_counter: 0, likes_counter: 0)
+    expect(post).to_not be_valid
+  end
 
-#  it 'is invalid with a long title' do
-#    post = Post.new(title: 'A' * 251, comments_counter: 0, likes_counter: 0)
-#    expect(post).to_not be_valid
-#  end
+  it 'is invalid with a long title' do
+    post = Post.new(title: 'A' * 251, comments_counter: 0, likes_counter: 0)
+    expect(post).to_not be_valid
+  end
 
-#  it 'is invalid with a negative comments_counter' do
-#    user = User.create(name: 'John', posts_counter: 0)
-#    post = Post.new(title: 'Test Post', comments_counter: -1, likes_counter: 0, author: user)
-#    expect(post).to_not be_valid
-#  end
+  it 'is invalid with a negative comments_counter' do
+    user = User.create(name: 'John', posts_counter: 0)
+    post = Post.new(title: 'Test Post', comments_counter: -1, likes_counter: 0, author: user)
+    expect(post).to_not be_valid
+  end
 
-#  it 'is invalid with a non-integer likes_counter' do
-#    user = User.create(name: 'John', posts_counter: 0)
-#    post = Post.new(title: 'Test Post', comments_counter: 0, likes_counter: 1.5, author: user)
-#    expect(post).to_not be_valid
-#  end
+  it 'is invalid with a non-integer likes_counter' do
+    user = User.create(name: 'John', posts_counter: 0)
+    post = Post.new(title: 'Test Post', comments_counter: 0, likes_counter: 1.5, author: user)
+    expect(post).to_not be_valid
+  end
 
-#  describe '#update_post_counter' do
-#    it 'increments the author\'s posts_counter when a post is saved' do
-#      user = User.create(name: 'John', posts_counter: 0)
-#      post = Post.new(title: 'Test Post', comments_counter: 0, likes_counter: 0, author: user)
+  describe '#update_post_counter' do
+    it 'increments the author\'s posts_counter when a post is saved' do
+      user = User.create(name: 'John', posts_counter: 0)
+      post = Post.new(title: 'Test Post', comments_counter: 0, likes_counter: 0, author: user)
 
-#      expect do
-#        post.save
-#        user.reload
-#      end.to change(user, :posts_counter).by(1)
-#    end
-#  end
-#end
+      expect do
+        post.save
+        user.reload
+      end.to change(user, :posts_counter).by(1)
+    end
+  end
+end
```

Comment:
- [ ] As required for the project to have tests to any custom methods in the models, kindly ensure that you also implement tests to the `#recent_comments` method in the `Post` model.

Reply:
I had already completed it before I repeated the module. Once my previous project is approved I have only to add tests for the models. Do you want to check the project requirement?

Reply:
@Gardimy 

But you didn't add tests for that method. Just add them and everything will be ok

Reply:
But I don't use this method in my post model.

Reply:
@Gardimy 

Just check all methods you have in the Post model, and ensure that you have tests to all of them.

Reply:
ok!